### PR TITLE
POC-569: Improve synced transcript auto-scroll anchoring

### PR DESCRIPTION
**Pull request by pocketcasts**
Resolves https://linear.app/a8c/issue/POC-569/improve-synced-transcript-auto-scroll-anchoring

## Summary
During synced transcript playback, the active highlighted cue would drift to the bottom of the screen because `scrollRangeToVisible` only scrolls the minimum amount needed. This PR anchors the active cue at ~30% from the top of the visible area for a consistent reading position.

## Changes
- Added a `verticalAnchor` parameter (default `0.5`) to `UITextView.scrollToRange(_:verticalAnchor:)` — existing callers (search navigation) keep centered behavior unchanged.
- Replaced `scrollRangeToVisible` in `updateTranscriptPosition()` with `scrollToRange(range, verticalAnchor: 0.3)`, removing the `length * 2` workaround.
- Updated `scrollBackToCurrentHighlight()` to use the same 30% anchoring so auto-scroll recovery matches normal playback.

## Verification
- **Lint**: PASS
- **Tests**: PASS — 318 tests, 0 failures
- **Diff review**: PASS — confirmed only intended changes, no debug artifacts
- **Secret scan**: PASS — no credentials in diff

## Confidence
**HIGH** — The change is minimal and surgical: one new default parameter in a UITextView extension, and two call-site updates. Build and all tests pass. Search navigation is unaffected since it uses the default `0.5` anchor. Edge cases (beginning/end of transcript) are handled by existing clamping logic.

---
This PR was created autonomously by linear-solver.
Triage complexity: medium | [Linear issue](https://linear.app/a8c/issue/POC-569/improve-synced-transcript-auto-scroll-anchoring)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/127">8.11 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/TranscriptViewController.swift [hunk 1]
```diff
@@ -841,8 +841,7 @@ class TranscriptViewController: PlayerItemViewController, AnalyticsSourceProvide
             previousRange = range
             transcriptView.attributedText = styleText(transcript: transcript, position: position)
             if !isUserScrolling, !isSearching, !isAutoScrollSuppressed {
-                let scrollRange = NSRange(location: range.location, length: range.length * 2)
-                transcriptView.scrollRangeToVisible(scrollRange)
+                transcriptView.scrollToRange(range, verticalAnchor: 0.3)
             }
```
[the remark this answers is not in the document]
Comment (leandroalonso) [reply podcasts/TranscriptViewController.swift]: Done in d151a80ec09bd042f5b5ef4678c0939daa0bec8b

File: podcasts/TranscriptViewController.swift [hunk 2]
```diff
@@ -922,7 +921,7 @@ class TranscriptViewController: PlayerItemViewController, AnalyticsSourceProvide
         // highlight is static and yanking the view back to it would fight the
         // user who deliberately scrolled elsewhere to read.
         guard !isSearching, !isUserScrolling, playbackManager.isPlayingEpisode, let previousRange else { return }
-        transcriptView.scrollToRange(previousRange)
+        transcriptView.scrollToRange(previousRange, verticalAnchor: 0.3)
     }
```
[the remark this answers is not in the document]
Comment (leandroalonso) [reply podcasts/TranscriptViewController.swift]: Done in d151a80ec09bd042f5b5ef4678c0939daa0bec8b

File: Modules/Sources/PocketCastsUtils/Extensions/UI/UITextView+scrollToRange.swift [hunk 3]
```diff
@@ -12,10 +12,9 @@ extension UITextView {
 
         let finalRect = rect.offsetBy(dx: textContainerInset.left, dy: textContainerInset.top)
 
-        // Calculate the rectangle to scroll to such that the finalRect is centered
         var visibleRect = CGRect(
             x: finalRect.origin.x - (bounds.width / 2) + (finalRect.width / 2),
-            y: finalRect.origin.y - (bounds.height / 2) + (finalRect.height / 2),
+            y: finalRect.origin.y - (bounds.height * verticalAnchor) + (finalRect.height / 2),
```
[the remark this answers is not in the document]
Comment (leandroalonso) [reply Modules/Sources/PocketCastsUtils/Extensions/UI/UITextView+scrollToRange.swift]: Done in b48cf3633c29fa20f6c98a9e8bd5c28e9fe149c0

**Comment by leandroalonso** [comment]
## Review feedback addressed

- **TranscriptViewController.swift:845** (@Copilot): extract duplicated `0.3` anchor into a named constant — Done in d151a80ec09bd042f5b5ef4678c0939daa0bec8b
- **TranscriptViewController.swift:925** (@Copilot): use shared anchor constant for auto-scroll recovery — Done in d151a80ec09bd042f5b5ef4678c0939daa0bec8b
- **UITextView+scrollToRange.swift:17** (@Copilot): clamp `verticalAnchor` to 0...1 — Done in b48cf3633c29fa20f6c98a9e8bd5c28e9fe149c0

**Review by leandroalonso — APPROVED** [review]
